### PR TITLE
[Technical] DynamicTableViewController improvements

### PR DESCRIPTION
**Pull request by eisvogel**
This PR improves the technical implementation of the `DynamicTableViewController` regarding its predefined cell types.

Cell types have been refactored from an enum type to a struct with static function defining the actually types. Support for the standard typography configurations as specified by the design guides has been added. The previous dynamic type cell type have been deprecated and will be removed after all view controllers have been migrated.

Furthermore a section shortcut for a navigation subtitle has been added as required by the most recent design updates regarding navigation bar titles and subtitles.

The risk legend has been modified to adopt these changes already, including some minor pixel spacing and color adjustments.

### CODE REVIEW

**Comment by EisvogelSAP** [comment]
Sorry, for the open/close bonfire. Something was wrong with the diff. Needed to convince GitHub to refresh it.

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/__tests__/DynamicTableViewModelTests.swift [hunk 1]
```diff
@@ -54,22 +54,20 @@ class DynamicTableViewModelTests: XCTestCase {
 		let section = sut.section(1)
 		
 		XCTAssertEqual(section.cells.count, 1)
-		let cell = section.cells.first
-		if case .bold(text: let text) = cell {
-			XCTAssertEqual(text, "Baz")
-		} else {
-			XCTFail()
-		}
+//		if let cell = section.cells.first {
```
Comment (melloskitten) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/__tests__/DynamicTableViewModelTests.swift:57]: can this be removed completely?

**Review by melloskitten — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewCell.swift [hunk 2]
```diff
@@ -38,24 +57,54 @@ enum DynamicCell {
...
+
+	static func title1(text: String, color: UIColor? = nil, configure: CellConfigurator? = nil) -> Self { .enaLabelStyle(.title1, text: text, color: color, configure: configure) }
+	static func title2(text: String, color: UIColor? = nil, configure: CellConfigurator? = nil) -> Self { .enaLabelStyle(.title2, text: text, color: color, configure: configure) }
+	static func headline(text: String, color: UIColor? = nil, configure: CellConfigurator? = nil) -> Self { .enaLabelStyle(.headline, text: text, color: color, configure: configure) }
+	static func body(text: String, color: UIColor? = nil, configure: CellConfigurator? = nil) -> Self { .enaLabelStyle(.body, text: text, color: color, configure: configure) }
+	static func subheadline(text: String, color: UIColor? = nil, configure: CellConfigurator? = nil) -> Self { .enaLabelStyle(.subheadline, text: text, color: color, configure: configure) }
+	static func footnote(text: String, color: UIColor? = nil, configure: CellConfigurator? = nil) -> Self { .enaLabelStyle(.footnote, text: text, color: color, configure: configure) }
+}
+
+extension DynamicCell {
```
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewCell.swift:100]: nice that you actually deprecated the methods!

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/__tests__/DynamicTableViewControllerTests.swift [hunk 3]
```diff
@@ -99,7 +99,7 @@ extension DynamicTableViewControllerTests {
 		let cell = sut.tableView.dequeueReusableCell(withIdentifier: reuseIdentifier,
 																								 for: IndexPath(row: 0, section: 0))
 		// assert type
```
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/__tests__/DynamicTableViewControllerTests.swift:101]: you could remove the assert type comment as it is pretty redundant now

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/__tests__/DynamicTableViewModelTests.swift (hunk 1 above)
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/__tests__/DynamicTableViewModelTests.swift:57]: dead code?

**Review by johannesrohwer — APPROVED** [review]
thanks for updating the dynamic table view! 👍 Just some minor things, however I will approve the PR and let it up to you if you want to change it.

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/__tests__/DynamicTableViewModelTests.swift (hunk 1 above)
Comment (EisvogelSAP) [reply src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/__tests__/DynamicTableViewModelTests.swift:57]: Well spotted. I missed that test. Will be updated shortly.

Comment (EisvogelSAP) [reply src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/__tests__/DynamicTableViewModelTests.swift:57]: Nope, only missed this one while updating.

**Review by melloskitten — APPROVED** [review]

**Pull request merged**